packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,286 @@
+
+![ ](spec-images/select-6.png)
+
+#### Icons
+
+The client-user should be able to specify whether they want left icons on list options or not. Icons show up in list options to the left of text and are displayed as part of the value in the field button.
+
+![ ](spec-images/select-7.png)
+
+#### Groups
+
+The client-user should be able to create groups for list options in the dropdown. Groups are not interactable. Filtering by group name should show all the options under that group, whether or not it is included in the option text.
+
+![ ](spec-images/select-8.png)
+
+#### Secondary Text
+
+> NOTE: NEEDS VISUAL DESIGN
+
+The client-user should be able to specify whether they want secondary text on list options or not. Filtering by secondary text should show all the options with that text.
+
+![ ](spec-images/select-9.png)
+
+The client-user should be able to specify whether they want other information (including groups) displayed as secondary text in the field input.
+
+![ ](spec-images/select-10.png)
+
+#### Selection
+
+> NOTE: NOT SURE THIS IS A REAL USE CASE FOR SELECT COMPONENT, SEE FUTURE CONSIDERATIONS FOR MULTI-SELECTION
+
+The client-user should be able to specify whether they want to display selection with or without a check mark.
+
+![ ](spec-images/select-11.png)
+
+#### Filter
+
+The client-user should be able to specify whether they want filtering available in the select component. The filter field appears in the dropdown. The filter applies to all list options, even ones that are on the server and aren't currently visible in the component.
+
+![ ](spec-images/select-12.png)
+
+### Behavior
+
+#### Minimum Width
+
+> NOTE: NEEDS DEV REVIEW
+
+The field button minimum width is three times the height of the component (to make components targetable on touch screens). The dropdown width is the same as the field button until reaching the minimum width (without truncating) of the longest string in the list options.
+
+![ ](spec-images/select-13.png)
+
+#### Dropdown Height
+
+The dropdown height should be enough to display five to seven list options before overflowing with a vertical scrollbar (not pictured).
+
+#### Overflow Text
+
+> NOTE: NEEDS DEV & VISUAL DESIGN REVIEW
+
+All text in the select component should truncate, not wrap.
+
+![ ](spec-images/select-14.png)
+
+#### Progress & Loading
+
+> NOTE: HOW THIS (AND ANY) FEATURE IS INCLUDED IN IMPLEMENTATION IS UP TO DEVELOPERS
+
+When displaying list options and the dropdown lags or is expected to take longer than normal, the client-user should be able to indicate loading via a slot.
+
+![ ](spec-images/select-15.png)
+
+> NOTE: IS THIS NEEDED? IF YES, NEEDS VISUAL DESIGN & TECH WRITER
+
+Dropdowns with dynamically loading list options may have a "Load more" button. Clicking the "Load more" button should resolve into the "Loading…" indication, then load more list options and scroll the dropdown as needed.
```

Comment:
Kinda generally for the dynamically load more options case, I'd be interested in knowing the workflow for:

1. Showing how to request more loads (this one covers that ✔)
2. Showing that more is loading (this one covers that ✔)
3. Showing that loading failed (should a message be shown? Is it dismissible? would there be only one? Should there be a retry interaction or just click load more again?)
4. How to cancel loading / when loading should be cancelled (should closing the pop-up cancel loading or should it allow loading and updating while closed? How can the user tell loading is done or failed or maybe they should just reopen it to check?)

Reply:
Erring on the side of low-impact UX, I'd probably recommend having the loading indicator turn back into the "load more" button after a reasonable amount of time, even if it's closed (if possible, there seem like there could be some edge cases here). I can add this to the spec but I actually can't remember whether @atmgrifter00 and I agreed I should have moved this to Future Considerations or not, because the expectation is that the client devs would implement this if they wanted, and we would not build it into the component.

Reply:
Moved to "Future Considerations"